sympy/physics/quantum/qapply.py

```diff
@@ -177,18 +190,40 @@ def qapply_Mul(e, **options):
         result = TensorProduct(*[qapply(lhs.args[n]*rhs.args[n], **options) for n in range(len(lhs.args))]).expand(tensorproduct=True)
         return qapply_Mul(e.func(*args), **options)*result
 
+    # For Sums, move the Sum to the right.
+    if isinstance(rhs, Sum):
+        if isinstance(lhs, Sum):
+            if set(lhs.variables).intersection(set(rhs.variables)):
+                raise ValueError('Duplicated dummy indices in separate sums in qapply.')
+            limits = lhs.limits + rhs.limits
+            result = Sum(qapply(lhs.function*rhs.function, **options), *limits)
+            if sum_doit:
+                result = result.doit()
+            return qapply_Mul(e.func(*args)*result, **options)
+        else:
+            result = Sum(qapply(lhs*rhs.function, **options), rhs.limits)
+            if sum_doit:
+                result = result.doit()
+            return qapply_Mul(e.func(*args)*result, **options)
+
+    if isinstance(lhs, Sum):
+        result = Sum(qapply(lhs.function*rhs, **options), lhs.limits)
+        if sum_doit:
+            result = result.doit()
+        return qapply_Mul(e.func(*args)*result, **options)
+
     # Now try to actually apply the operator and build an inner product.
     try:
         result = lhs._apply_operator(rhs, **options)
-    except NotImplementedError:
+    except (NotImplementedError, AttributeError):
         result = None
```

Comment:
In general I think it is better if we don't catch exceptions like `AttributeError`. It can easily mask bugs and other things and is not really a good way of doing control flow.

What is this `AttributeError` being raised by? If the exception is being raised then does that not really indicate a bug elsewhere like some type is missing a method or something?

Reply:
Fixed in a commit this morning. Thanks!